kind/common.sh

```diff
@@ -75,48 +75,54 @@ function _get_prometheus_operator_images {
         grep -R "image:" kube-prometheus/manifests/*grafana* | awk '{print $3}'
     fi
 }
 
 function _load_prometheus_operator_images_to_local_registry {
     for img in $(_get_prometheus_operator_images); do
-        $CTR_CMD pull $img
-        $KIND load docker-image $img
+        $CTR_CMD pull "$img"
+        $KIND load docker-image "$img"
     done
 }
 
 function _fetch_microshift {
     # pulls the image from quay.io
-    $CTR_CMD pull ${MICROSHIFT_IMAGE}:${MICROSHIFT_TAG}
+    $CTR_CMD pull "${MICROSHIFT_IMAGE}":"${MICROSHIFT_TAG}"
 }
 
 function _wait_microshift_up {
     # wait till container is in running state
-    while [ "$(${CTR_CMD} inspect -f '{{.State.Status}}' ${MICROSHIFT_CONTAINER_NAME})" != "running" ];do
+    while [ "$(${CTR_CMD} inspect -f '{{.State.Status}}' "${MICROSHIFT_CONTAINER_NAME}")" != "running" ]; do
         echo "Waiting for container ${MICROSHIFT_CONTAINER_NAME} to start..."
         sleep 5
     done
     echo "Container $MICROSHIFT_CONTAINER_NAME} is now running!"
 
     echo "Waiting for cluster to be ready ..."
 
-    while [ -z "$($CTR_CMD exec --privileged ${MICROSHIFT_CONTAINER_NAME} kubectl --kubeconfig=/var/lib/microshift/resources/kubeadmin/kubeconfig get nodes -o=jsonpath='{.items..status.conditions[-1:].status}' | grep True)" ]; do
+    while [ -z "$($CTR_CMD exec --privileged "${MICROSHIFT_CONTAINER_NAME}" \
+        kubectl --kubeconfig=/var/lib/microshift/resources/kubeadmin/kubeconfig \
+        get nodes -o=jsonpath='{.items..status.conditions[-1:].status}' | grep True)" ]; do
         echo "Waiting for microshift cluster to be ready ..."
         sleep 20
     done
 }
 
 function _deploy_microshift_cluster {
+    # create network for microshift and registry to communicate
+    $CTR_CMD network create ${DEFAULT_NETWORK}
     # run the docker container
-    $CTR_CMD run -d --name ${MICROSHIFT_CONTAINER_NAME} --privileged -v microshift-data:/var/lib -p 6443:6443 -p 80:80 -p 443:443 ${MICROSHIFT_IMAGE}:${MICROSHIFT_TAG}
+    $CTR_CMD run -d --name "${MICROSHIFT_CONTAINER_NAME}" --privileged \
+        -v microshift-data:/var/lib -p 6443:6443 -p 80:80 -p 443:443 --network ${DEFAULT_NETWORK} "${MICROSHIFT_IMAGE}":"${MICROSHIFT_TAG}"
 }
 
 function _deploy_prometheus_operator {
-    git clone -b ${PROMETHEUS_OPERATOR_VERSION} --depth 1 https://github.com/prometheus-operator/kube-prometheus.git
-    sed "s/replicas: 2/replicas: ${PROMETHEUS_REPLICAS}/g" kube-prometheus/manifests/prometheus-prometheus.yaml | tee -a kube-prometheus/manifests/prometheus-prometheus.yaml > /dev/null
-    if [ ${PLATFORM} == "kind" ]; then
+    git clone -b "${PROMETHEUS_OPERATOR_VERSION}" --depth 1 https://github.com/prometheus-operator/kube-prometheus.git
+    sed "s/replicas: 2/replicas: ${PROMETHEUS_REPLICAS}/g" kube-prometheus/manifests/prometheus-prometheus.yaml | \
+        tee -a kube-prometheus/manifests/prometheus-prometheus.yaml >/dev/null
```

Comment:
why need a empty tee? or tree?

Reply:
This was done because of different implementations of sed command on MacOS. Since `sed -i -e` was creating a file with `-e` extension so instead of using `sed -i -e` I used `tee` which will write sed command stdout to a file. Hence not creating any new file with `-e` extension.